src/Concretize.hs

```diff
@@ -87,8 +87,8 @@ concretizeXObj allowAmbiguityRoot typeEnv rootEnv visitedDefinitions root =
         let functionEnv = Env Map.empty (Just env) Nothing Set.empty InternalEnv 0
             envWithArgs =
               foldl'
-                ( \e arg@(XObj (Sym (SymPath _ argSymName) _) _ _) ->
-                    extendEnv e argSymName arg
+                ( \e arg@(XObj (Sym path _) _ _) ->
+                    let Right v = insertX e path arg in v
```

Comment:
Are you sure this never fails? (seems kind of off to have an incomplete pattern among all the new safety..!)

Reply:
Yes. It will never fail because:

1. We are inserting into a brand new functionEnv, declared above.
2. We are inserting binders for the functions arguments, which *must* be unqualified (checked elsewhere).

Though I agree it'd be better to do this in a safe way. I've added a comment to this effect for now.